File: src/main/java/baseDatos/ProductoPersistente.java
package baseDatos;

import gestorAplicacion.ventas.Producto;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectOutputStream;
import java.util.List;

/**
 * Clase que implementa la interfaz Persistente para manejar la escritura y lectura de los productos en archivos.
 *
 * @author Juan Daniel Bula, Leidy Daniela Alzate, Oscar Andres Usuga
 */
public class ProductoPersistente implements Persistente<Producto> {

    @Override
    public List<Producto> leerTodos() {
        return null;
    }

    @Override
    public void guardar(Producto producto) {
        FileOutputStream fos;
        try {
            String ruta = System.getProperty("user.dir") + String.format("\\src\\main\\java\\baseDatos\\temp\\productos\\%s.txt", producto.getCodigo());
            File file = new File(ruta);
            file.createNewFile();
            fos = new FileOutputStream(file);
            ObjectOutputStream oos = new ObjectOutputStream(fos);
            oos.writeObject(producto);
            oos.close();
            fos.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    @Override
    public Producto leerUno(String identificador) {
        return null;
    }
}

File: python/basedatos/producto_persistente.py
import os
import pickle


class ProductoPersistente:

    def __init__(self):
        self._path = f'{os.path.dirname(os.path.abspath(__file__))}/temp/productos'

    def guardar(self, producto):
        file = f'{self._path}/{producto.get_codigo()}'
        try:
            with open(file, 'wb') as f:
                pickle.dump(producto, f, pickle.HIGHEST_PROTOCOL)
        except Exception:
            raise FileNotFoundError()

    def leer_uno(self, identificador):
        file = f'{self._path}/{identificador}'
        try:
            with open(file, 'rb') as f:
                producto = pickle.load(f)
                return producto
        except Exception:
            raise FileNotFoundError()
